### DISCUSSION TITLE
How to setup ESP correctly with one Router and two repeaters in mesh

### DISCUSSION THREAD

**Discussion by Kuto587** [jomjol/AI-on-the-edge-device#2316] [Q&A]
Hello everyone,

I am trying to use the Ai-on-the-edge-device, however I find it difficult to connect to it via the webinterface.
My network setup is as follows:

1. One FritzBox! Router in the top floor (192.168.178.1)
2. One repeater on ground level            (192.168.178.21)
3. One repeater in the basement           (192.168.178.23) -> The Watermeter connects to this router and was assigned the Ip-Adress 192.168.178.20
All these devices are connected in a Mesh-network.

However I cannot connect to the watermeter with my Browser. From the Webinterface of the Router, I see that it connected to the network, but still I cannot reach it.

I then explicitly wrote the IP-Adress which the ESP got from the router into the wlan.ini file. I then can connect to the ESP but only if I am using the same Repeater to access it, and I write that Repeater´s IP-Adress into the gateway filed of the ini file

So my question is: Why can´t it Access the throughout the network? How shall I configure the wlan.ini file such that I can access the ESP from the whole network? Which Gateway shall I use? 

I hope I explained it somewhat cleary, thanks for your help!
Best,
Phil

**Comment by flyingToast3r** [comment]
The problem that the web server connection works only over the same AP (repeater) sounds really strange. Sometimes, e.g., if the signal is weak the web server responds a bit slowly, and you had just once "luck" that the connection worked in the moment where your client used the same AP. Try a ping when your client device is connected to one of the other APs. 
Another hint could be the singal strenght that is reported on the "overview" page of the web server. If the connection is proper then you can continue optimizing your network configuration otherwise the issue may lies in the weak physical connection.

**Comment by caco3** [comment]
mesh support is still experimental.
do you see to which AP it effectively connects? maybe it only connects to your Fritzbox and ignores the repeaters.

you could enable AP scanning and see if that brings an improvement: https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-rssithreshold

**Reply by caco3** [reply]
siehe auch https://github.com/jomjol/AI-on-the-edge-device/discussions/2329

**Reply by Kuto587** [reply]
Well, I never had any luck connecting the device to the mesh network. 
In the end, I had to move the router anyways and now the device can directly connect to the FritzBox. Now it works perfectly.

So to sum up: I never had good results with connecting this device over repeaters no matter the configuration...

**Comment by friedpa** [comment]
I had the same issue with a FritzRouter and two FritzRepeater in the Ground Floor of my house. You should try to ping the Repeaters. If you see more than 50ms response time (as a mximum) then just use a Devolo Magic (with WIFI). This brought the latency down to 3 ms and the problem disappeared. Use "_ping -t -l 1024 IP-Address-of-the-Repeater_" for testing.
